### PR TITLE
Fix: golangci lint

### PR DESCRIPTION
**Pull request by 0luwaj0x1**
This PR fixes [Issue #706](https://github.com/argoproj-labs/argocd-image-updater/issues/706)

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -15,5 +11,7 @@ linters:
 linters-settings:
   goimports:
     local-prefixes: github.com/argoproj-labs/argocd-image-updater
-service:
-  golangci-lint-version: 1.52.2
+issues:
+  exclude:
+    - ".*\\.pb\\.go"
+    - "vendor/.*"
```
Comment (chengfang) [inline .golangci.yml:17]: I think we should list `exclude-files` and `exclude-dirs` separately for clarity. In the above the vendor exclusion doesn't seem to include all subdirs.

```yaml
exclude-files:
  - ".*\\.pb\\.go"
exclude-dirs:
  - vendor/
```

File: .github/workflows/ci-tests.yaml [hunk 2]
```diff
@@ -59,10 +59,12 @@ jobs:
...
         uses: actions/checkout@v3
+      - name: Setup Golang
+        uses: actions/setup-go@v3
       - name: Run golangci-lint
-        uses: golangci/golangci-lint-action@v3
+        uses: golangci/golangci-lint-action@v5
         with:
-          version: v1.52.2
+          version: v1.57.2
           args: --timeout 5m
```
Comment (chengfang) [inline .github/workflows/ci-tests.yaml:68]: We're still seeing lint failures, though with a slightly better error message. Wondering if https://github.com/golangci/golangci-lint/issues/395#issuecomment-1690510780 could help?

File: Dockerfile [hunk 3]
```diff
@@ -14,7 +14,9 @@ FROM alpine:latest
 
 RUN apk update && \
     apk upgrade && \
-    apk add ca-certificates git openssh-client python3 py3-pip tini && \
+    apk add --no-cache ca-certificates git openssh-client python3 py3-pip tini && \
+    python3 -m venv /venv && \
+    . /venv/bin/activate && \
```
Comment (chengfang) [inline Dockerfile:19]: #700 already has a fix for the above problem in Dockerfile. I think we should treat this as a separate issue to be addressed by #700. If you agree, can you please undo this change in Dockerfile, and keep this PR for fixing lint? @0luwaj0x1

**Comment by chengfang** [comment]
@0luwaj0x1 can you also squash all commits?

File: Dockerfile (hunk 3 above)
Comment (0luwaj0x1) [reply Dockerfile:19]: Agreed

**Review by chengfang — APPROVED** [review]

**Pull request merged**